Login with username

shridarpatil opened the pull request:
Use a username to login
https://github.com/frappe/frappe/issues/4106

shridarpatil committed:
Login with username

https://github.com/frappe/frappe/issues/4106

felixvarghese commented:
If possible, on the login screen, change the placeholder text to "Email Address or Username"

shridarpatil committed:
Change placeholder to 'Email address or Username' in login screen

    Change placeholder in login screen based on system setting
    If allowed login using mobile number placeholder => 'Email address or Mobile number'
    If allowed login using username placeholder => 'Email address or Username'
    If allowed login using both username and mobile placeholder => 'Email Address or Mobile number or Username'

shridarpatil committed:
Merge branch 'develop' into develop

rmehta commented on frappe/www/login.html:
```diff
@@ -17,9 +17,16 @@
-					_('Email address or Mobile number')
-					if frappe.utils.cint(frappe.db.get_value('System Settings', 'System Settings', 'allow_login_using_mobile_number'))
-					else _('Email address') }}"
+					_('Email address or Mobile number or Username')
```
@shridarpatil thanks for contributing!

this seems very hard to maintain / debug.

How about adding the placeholder in the `get_context` method in `login.py` so that the code looks much cleaner. Let me know if its not clear

shridarpatil committed:
Change place holder based on system settings

add place holder in login.py and then use in template

rmehta committed:
Update login.py

shridarpatil committed:
Removed trailing whitespaces from login.py
